### PR TITLE
Split the timer on/off part of the overlay

### LINKED ISSUE #1758: ability to turn on/off different times rather then all or nothing
reasoning: if the user is only interested in the remaining turn timer, or their total time spent

current implementation: enable/disable all timers with no independence

### PR DESCRIPTION
**Pull request by polohb**
- Opponent, Player and Turn are 3 separated choices in the option menu
- Shortcut cycle to show/hide each item
- fixes #1758

### CODE REVIEW

**Comment by Epix37** [comment]
[Please use tabs for formatting](https://github.com/HearthSim/Hearthstone-Deck-Tracker/blob/master/CONTRIBUTING.md).

File: Hearthstone Deck Tracker/Utility/HotKeys/PredefinedHotKeyActions.cs [hunk 1]
```diff
@@ -82,13 +82,72 @@ public static void ToggleOverlaySecrets()
...
+                    Config.Instance.HideOpponentTimer = true;
+                    Config.Instance.HidePlayerTimer = true;
+                    Config.Instance.HideTurnTimer = true;
+                    break;
+                default:
+                    break;
+            }
+            if (Config.Instance.HideTimerCycle == 7) Config.Instance.HideTimerCycle = 0;
+            else Config.Instance.HideTimerCycle++;
+
```
Comment (Epix37) [inline Hearthstone Deck Tracker/Utility/HotKeys/PredefinedHotKeyActions.cs]: You can write this as 

```C#
var timerState = Convert.ToInt32(Config.Instance.HideOpponentTimer) << 2
				| Convert.ToInt32(Config.Instance.HidePlayerTimer) << 1
				| Convert.ToInt32(Config.Instance.HideTurnTimer);
timerState = timerState >= 7 ? 0 : timerState + 1;
Config.Instance.HideOpponentTimer = (timerState & 4) > 0;
Config.Instance.HidePlayerTimer = (timerState & 2) > 0;
Config.Instance.HideTurnTimer = (timerState & 1) > 0;
```

This also allows you to get rid of the `HideTimerCycle` config value entirely.

**Comment by polohb** [comment]
Hi, thx for the comments.

This is my first contribution to a public project, maybe  I don't have done things in the right order, I will try to first go on gitter or irc to take some advice.

For the tabs, I leave VisualStudio with default options, I have to look at this.

**Comment by Epix37** [comment]
One problem I see is that this will reset peoples timer settings (i.e., timer being hidden). The only solution I can think of right now is to leave the `HideTimers` config value in, marked as obsolete, and transfer the value in `ConfigManager.ConvertLegacyConfig`. Any better ideas?

**Comment by Jonysegal** [comment]
Closing due to it being five years old.

**Comment by Jonysegal** [comment]
Closed due to it being 5 years old.

**Pull request closed without merging**